doc/source/howtos/encrypt-traffic.rst

```diff
@@ -0,0 +1,307 @@
+
+  .. note:: Whatever method you use to generate the certificate and key files,
+     the ``Common Name`` value used for the server and client certificates/keys
+     must each differ from that value used for the CA certificate.
+     Otherwise, the certificate and key files will not work for servers
+     compiled using OpenSSL.
+
+     The easiest way to do this is to give the CA file a common name (CN)
+     as follows::
+
+      openssl req -new -x509 -nodes -days 3600 -key ca-key.pem -out ca.pem -subj “/CN=my_cluster_name”
+
+.. note:: Percona XtraBackup supports keyring transfer
+   in version 2.4.4 and later.
+
+.. note:: SSL clients require DH parameters to be at least 1024 bits.
+   However, versions of ``socat`` earlier than 1.7.3 use 512-bit parameters.
+   If a :file:`dhparams.pem` file of required length is not found during SST
+   in the data directory,
+   it is generated with 2048 bits, which can take several minutes.
+   To avoid this delay, create the :file:`dhparams.pem` file manually
+   and place it in the data directory before joining the node to the cluster.
+
+IST Traffic, Write-Set Replication, and Service Messages
+========================================================
+
+IST refers to transferring only missing transactions from DONOR to JOINER node.
+Write-set replication is the main workload in |PXC|
+whenever a transaction is performed on one node,
+it is replicated to all other nodes.
+Service messages ensure that all nodes are synchronized.
+
+All of this traffic is transferred via the same underlying communication
+channel used by Galera (``gcomm``).
+Securing this channel will ensure that IST traffic, write-set replication,
+and service messages are encypted.
+
+To enable SSL for all internal node processes,
+define the paths to the key, certificate and certificate authority files
+using the following parameters:
+
+* |socket.ssl_key|_
+* |socket.ssl_cert|_
+* |socket.ssl_ca|_
+
+.. |socket.ssl_key| replace:: ``socket.ssl_key``
+.. _socket.ssl_key: http://galeracluster.com/documentation-webpages/galeraparameters.html#socket-ssl-key
+
+.. |socket.ssl_cert| replace:: ``socket.ssl_cert``
+.. _socket.ssl_cert: http://galeracluster.com/documentation-webpages/galeraparameters.html#socket-ssl-cert
+
+.. |socket.ssl_ca| replace:: ``socket.ssl_ca``
+.. _socket.ssl_ca: http://galeracluster.com/documentation-webpages/galeraparameters.html#socket-ssl-ca
+
+To set these parameters, use the :variable:`wsrep_provider_options` variable.
+
+.. code-block:: text
+
+   wsrep_provider_options="socket.ssl=yes;socket.ssl_key=/path/to/server-key.pem;socket.ssl_cert=/path/to/server-cert.pem;socket.ssl_ca=/path/to/cacert.pem"
+
+For more information, see `Index of wsrep provider options <https://www.percona.com/doc/percona-xtradb-cluster/5.7/wsrep-provider-index.html>`_.
+
+.. note:: You must use the same key and certificate files on all nodes,
+   preferably those used for :ref:`client-server-encryption`.
```

Comment:
*There are 2 set of 2 certificates in client-server encryption which one does user copy over.
We need to be specific.

* For encrypt=4 option too we can claim the same statement

* Also, user can copy over encrypt=4 generated cert here.